data/query.go

```diff
@@ -120,6 +204,115 @@ func GetSearchAPIQuery(validatedQueryParams SearchURLParams, censusTopicCache *c
+	apiQuery := createSearchAPIQuery(validatedQueryParams)
+	var contentTypes = ""
+	switch template {
+	case "all-adhocs":
+		contentTypes = "static_adhoc"
+	case "all-methodologies":
+		contentTypes = "static_qmi," + "static_methodology," + "static_methodology_download"
+	case "home-datalist":
+		contentTypes = "static_adhoc," + "timeseries," + "dataset_landing_page," + "reference_tables"
+	case "home-publications":
+		contentTypes = "bulletin," + "article," + "article_download," + "compendium_landing_page"
+	case "published-requests":
+		contentTypes = "static_foi"
+	case "home-list":
+		contentTypes = "static_page"
+	case "home-methodology":
+		contentTypes = "static_qmi," + "static_methodology," + "static_methodology_download"
+	case "time-series-tool":
+		contentTypes = "timeseries"
+	}
+
+	if apiQuery.Get("content_type") == "" {
+		apiQuery.Set("content_type", contentTypes)
+	}
+
+	return apiQuery
+}
+
+// GetDates finds the date from and date to parameters
+func GetDates(ctx context.Context, params url.Values) (startDate, endDate Date, err error) {
+	var (
+		startTime, endTime time.Time
+	)
+
+	const (
+		DayBefore   = "before-day"
+		DayAfter    = "after-day"
+		MonthBefore = "before-month"
+		MonthAfter  = "after-month"
+		YearBefore  = "before-year"
+		YearAfter   = "after-year"
+		DateFrom    = "fromDate"
+		DateTo      = "toDate"
+	)
+
+	yearAfterString, monthAfterString, dayAfterString := params.Get(YearAfter), params.Get(MonthAfter), params.Get(DayAfter)
+	yearBeforeString, monthBeforeString, dayBeforeString := params.Get(YearBefore), params.Get(MonthBefore), params.Get(DayBefore)
+	logData := log.Data{
+		"year_after": yearAfterString, "month_after": monthAfterString, "day_after": dayAfterString,
+		"year_before": yearBeforeString, "month_before": monthBeforeString, "day_before": DayBefore,
+	}
+
+	startTime, err = getValidTimestamp(yearAfterString, monthAfterString, dayAfterString)
+	if err != nil {
+		log.Warn(ctx, "invalid date, startDate", log.FormatErrors([]error{err}), logData)
+		return Date{}, Date{}, err
+	}
+
+	startDate = DateFromTime(startTime)
+
+	endTime, err = getValidTimestamp(yearBeforeString, monthBeforeString, dayBeforeString)
+	if err != nil {
+		log.Warn(ctx, "invalid date, endDate", log.FormatErrors([]error{err}), logData)
+		return Date{}, Date{}, err
+	}
+
+	endDate = DateFromTime(endTime)
+
+	if !startTime.IsZero() && !endTime.IsZero() && startTime.After(endTime) {
+		log.Warn(ctx, "invalid date range: start date after end date", log.Data{DateFrom: startDate, DateTo: endDate})
+		return Date{}, Date{}, errors.New("invalid dates: start date after end date")
+	}
+
+	return startDate, endDate, nil
+}
+
+func getValidTimestamp(year, month, day string) (time.Time, error) {
+	if year == "" || month == "" || day == "" {
+		return time.Time{}, nil
+	}
+
+	y, err := yearValidator(year)
+	if err != nil {
+		log.Error(context.TODO(), "Invalid parameter", err)
```

Comment:
Not sure if I saw this first time round 🤔 
```suggestion
		log.Error(ctx, "Invalid parameter", err)
```